src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -208,6 +209,7 @@ function WebGLShadowMap( _renderer, _lights, _objects, capabilities ) {
 
 			_lightPositionWorld.setFromMatrixPosition( light.matrixWorld );
 			shadowCamera.position.copy( _lightPositionWorld );
+			_shadowCameraFar = (light.shadow && light.shadow.camera.far) || light.distance || 1000;
```

Comment:
That's how it was in my first draft. It's just that, if I understand correctly, `light.distance` also controls attenuation and sometimes you might want to have the shadow camera far distance to be closer than the light distance, to get better shadow resolution.